### PR TITLE
Fix rosbag2_tests resource files and play_end_to_end test

### PR DESCRIPTION
**Pull request by mabelzhang**
Re-recorded the `cdr_test` and `wrong_rmw_test` bag files in `rosbag2_tests` with the latest `master` branch, metadata version 4. 
Updated `play_end_to_end` test to reflect the changes and re-enabled this test in CMakeLists.txt.

Only change in the bag is that the field `test_msgs::msg::BasicTypes::string_value` doesn't exist anymore, so it is replaced with a `int32_value`.
The messages are produced with a simple script [here](https://gist.github.com/mabelzhang/0b8df9224b13893d4b856a53506a95e8).

### CODE REVIEW

File: rosbag2_tests/test/rosbag2_tests/test_rosbag2_info_end_to_end.cpp [hunk 1]
```diff
@@ -42,20 +42,21 @@ TEST_F(InfoEndToEndTestFixture, info_end_to_end_test) {
...
-      "\nFiles:             cdr_test\\.db3"
+      "\nFiles:             cdr_test_0\\.db3"
       "\nBag size:          .*B"
       "\nStorage id:        sqlite3"
-      "\nDuration:          0\\.155s"
-      "\nStart:             Sep 18 2018 .*:.*:44.241 \\(1537282604\\.241\\)"
-      "\nEnd                Sep 18 2018 .*:.*:44.397 \\(1537282604\\.397\\)"
+      "\nDuration:          0\\.151s"
+      "\nStart:             Apr  9 2020 .*:.*:36.763 \\(1586406456\\.763\\)"
+      "\nEnd                Apr  9 2020 .*:.*:36.914 \\(1586406456\\.914\\)"
```
Comment (Karsten1987) [inline rosbag2_tests/test/rosbag2_tests/test_rosbag2_info_end_to_end.cpp:50]: One thing which bothered me constantly was that `End` is not ending with a `:`. 
Do you mind adding this? It's a very simple fix (the change has to be made in `rosbag2_transport`.)

Comment (mabelzhang) [reply rosbag2_tests/test/rosbag2_tests/test_rosbag2_info_end_to_end.cpp]: d12c761

**Comment by mabelzhang** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10049/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5647/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8178/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9959/)

**Comment by mabelzhang** [comment]
Updated .repos file. Retry:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10050/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5649/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8179/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9960/)

**Comment by mabelzhang** [comment]
Rebased and new set of CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10056/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5654/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8183/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9964/)

Actually it looks like the version run with CI above has already been rebased, for I only forked this branch today, and the fix was yesterday. I had to update my workspace to build with `ament_export_targets`. That was before I ran the first set of CI here.

So I don't know where that warning is coming from. Not from this branch - grep doesn't even show `ament_export_interfaces` anymore.

I'm looking at all the CIs here, and there hasn't been one green. I clicked on a few random ones, and they all have the same warning. Maybe there is one repo that hasn't been updated with the new command.
https://ci.ros2.org/job/ci_linux/

**Comment by Karsten1987** [comment]
That's a known issue. There's an outstanding PR for tracing tools to address the deprecation warning.

**Comment by Karsten1987** [comment]
Looking at CI, I think the `play` test doesn't pass neither on Linux nor on Windows.

**Comment by mabelzhang** [comment]
No more test failures on Linux & macOS.
Windows tests fail on record_end_to_end. Is that new?

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10080/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5667/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8198/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9982/)

**Comment by piraka9011** [comment]
@mabelzhang yes that's a known issue on Windows.
Splitting + Compression E2E tests are flaky since there's no support for ctrl-c (SIGINT) preventing the rosbag2 node from shutting down properly and cleaning up.

This solution is to refactor the E2E tests to use launch_testing (Some of the work started in #351 and #347).

**Pull request merged**